### PR TITLE
RHS info cards support

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Add support for add Info cards to the RHS section

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Needs to be implemented in mobile (link to PR or User Story)
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates

### CODE REVIEW

**Comment by jespino** [comment]
Here is a quick and dirty prototype, I have to review my own code and add the unit tests. But more or less implements what I expect for a first release.

**Comment by hanzei** [comment]
Thank you very much for the explanation!

**Comment by cpanato** [comment]
@jespino there is anything on the server side that needs to be done to support this?

**Comment by jespino** [comment]
Nope, is all based on already existing post message props field which contains arbitrary data

**Comment by jespino** [comment]
There are two things that needs to be solved before, close the RHS when the current post is deleted, and probably would be great to have a better suggestion about the "icon" from @michaelgamble

**Comment by cpanato** [comment]
@jespino you set this in the PR
> Needs to be implemented in mobile (link to PR or User Story)
There is a PR for mobile or are you waiting for more feedback?

Also, is it possible to add/write some documentation on how to use the extra info?

**Comment by jespino** [comment]
@cpanato Yes, I'm waiting for get the approval here before I create a US for mobile. And the same for the documentation. I'm not sure if the current approach is the correct one, maybe @levb and @crspeller wants to make changes to the field name, or maybe they want to take a complete different approach on int, so... I'm waiting until they review it :)

**Comment by jespino** [comment]
@jasonblais @asaadmahmood how we unblock this?

**Comment by jasonblais** [comment]
> how we unblock this?

@jespino How do you mean? Do you need something from us to move it forward? Or would like someone to test the changes?

**Comment by jespino** [comment]
@jasonblais Yes, I think this needs a second review, from @levb in this case, and maybe the final approval of what is implemented, probably from you. And I don't need if @asaadmahmood needs to touch anything else.

**Comment by jasonblais** [comment]
@jespino Maybe we can do a quick 15-minute screenshare? Or demo on Thursday's R&D meeting? That might be most efficient for PM approval.

I'll DM you

**Comment by jespino** [comment]
@jasonblais Done! :)

**Comment by jespino** [comment]
@jasonblais you can see the changes now in the spinmint. I added some example posts.

**Comment by jespino** [comment]
@jasonblais The first problem is only in compact view and I added a message to the spinmint with a very long username, everything looks like working, except the footer of the RHS Card view.

@asaadmahmood can you take care of both problems? looks like CSS related.

**Comment by asaadmahmood** [comment]
@jespino Will do.

**Comment by asaadmahmood** [comment]
@jasonblais For the second one, it works fine.
Screenshot 2019-06-14 at 8 33 38 PM

File: components/rhs_card_header/rhs_card_header.jsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        const expandSidebarTooltip = (
+            <Tooltip id='expandSidebarTooltip'>
+                <FormattedMessage
+                    id='rhs_header.expandSidebarTooltip'
+                    defaultMessage='Expand Sidebar'
+                />
+            </Tooltip>
+        );
+
+        const shrinkSidebarTooltip = (
```
[the remark this answers is not in the document]
Comment (jespino) [reply components/rhs_card_header/rhs_card_header.jsx]: Yes, I agree with that, but is following the existing pattern and that refactor should be part of another PR. I'm more than happy of file a new user story to generate a shared component for this header.

**Comment by asaadmahmood** [comment]
@jespino @jasonblais Fixed the other one too.
Screenshot 2019-06-14 at 8 37 25 PM

**Comment by asaadmahmood** [comment]
Don't merge, just submitting one quick fix.

**Comment by asaadmahmood** [comment]
Also fixed the long name in the footer.
Screenshot 2019-06-14 at 9 49 29 PM

**Comment by jespino** [comment]
## :fireworks:

**Comment by jasonblais** [comment]
Woot!! This is awsome

**Comment by aaronrothschild** [comment]
@amyblais I spoke with @jespino and he would like this to be undocumented until the mobile part is completed and rolled out, likely in 5.16.  What should I label this PR as?

**Comment by amyblais** [comment]
Changed milestone to v5.16

**Comment by aaronrothschild** [comment]
This has been documented in the webhooks docs: https://developers.mattermost.com/integrate/incoming-webhooks/#parameters

**Pull request merged**